modules/product/src/main/java/com/opengamma/strata/product/SecurityTrade.java

```diff
@@ -5,97 +5,67 @@
- * managed as reference data, separately from the trade.
+ * This trade represents a trade in a security, defined by a quantity and price.
+ * The security is embedded directly, however the underlying product model is not available.
+ * The security may be of any kind, including equities, bonds and exchange traded derivative (ETD).
  * <p>
  * Implementations of this interface must be immutable beans.
- * 
- * @param <P>  the type of the product
  */
-public interface SecurityTrade<P extends Product>
-    extends FinanceTrade, LinkResolvable<SecurityTrade<P>> {
+public interface SecurityTrade
+    extends FinanceTrade {
 
   /**
-   * Gets the link to the security that was traded.
-   * <p>
-   * A security link provides loose coupling between different parts of the object model.
-   * For example, this allows the trade to be stored in a different database to the security.
-   * <p>
-   * A link can be in one of two states, resolvable and resolved.
-   * In the resolvable state, the link contains the identifier and product type of the security.
-   * In the resolved state, the link directly embeds the security.
-   * <p>
-   * When the link is in the resolvable state, the {@link #getSecurity()} AND {@link #getProduct()}
-   * methods will throw an {@link IllegalStateException}.
-   * <p>
-   * To ensure that the link is resolved, call {@link #resolveLinks(LinkResolver)}.
+   * Gets the security that was traded.
    * 
-   * @return the link to the security, which may be in either the resolvable or resolved state
+   * @return the security
    */
-  public abstract SecurityLink<P> getSecurityLink();
+  public abstract Security getSecurity();
 
   /**
-   * Gets the security that was traded, throwing an exception if not resolved.
+   * Gets the quantity that was traded.
    * <p>
-   * Returns the underlying security that was traded.
-   * This is obtained from {@link #getSecurityLink()}.
-   * <p>
-   * The link has two states, resolvable and resolved.
-   * The security can only be returned if the link is resolved.
-   * If the link is in the resolvable state, this method will throw an {@link IllegalStateException}.
-   * <p>
-   * To ensure that the link is resolved, call {@link #resolveLinks(LinkResolver)}.
+   * This will be positive if buying and negative if selling.
    * 
-   * @return full details of the security
-   * @throws IllegalStateException if the security link is not resolved
+   * @return the quantity
    */
-  public default Security<P> getSecurity() {
-    return getSecurityLink().resolvedTarget();
-  }
+  public abstract long getQuantity();
 
   /**
-   * Gets the underlying product that was agreed when the trade occurred, throwing an exception if not resolved.
-   * <p>
-   * Returns the underlying product that captures the contracted financial details of the trade.
-   * This is obtained from {@link #getSecurityLink()}.
+   * Gets the price that was traded.
```

Comment:
This sounds odd to me. Do you trade a price? "The price at which the trade was executed" or something like that.